### PR TITLE
feat: add favourite icon toggler lab

### LINKED ISSUE #450: chore: add lab for DOM and events project
We have a workshop and lecture videos to teach campers about the DOM and click events.

Here is the preceding workshop
https://github.com/freeCodeCamp/CurriculumExpansion/pull/449

Campers will have already complete the previous fundamentals modules(strings, variables, arrays, numbers, Math object, objects, etc) So you can see the sheet for more details to see the complete list of what is covered.

For this lab, we want them to practice click events and updating styles with JavaScript.

NOTE: they will not have learned about change or submit events. So this lab should not include that. 

## Acceptance criteria

- [ ] create a new folder of the project in the `frontend-cert/js-projects` directory
- [ ] add your completed project in an `index.html`, `styles.css` and `script.js` files in that new folder
- [ ] create a markdown file for your proposed user stories. You should call it `user-stories.md` and place it in your new project folder

### PR DESCRIPTION
**Pull request by zairahira**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #450

### CODE REVIEW

File: frontend-cert/js-projects/lab-favourite-toggler/user-stories.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+You should have an `h1` heading that contains the a heading.
```
Comment (Dario-DC) [inline frontend-cert/js-projects/lab-favourite-toggler/user-stories.md:1]:
```suggestion
1. You should have an `h1` element that contains the heading.
```

File: frontend-cert/js-projects/lab-favourite-toggler/user-stories.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+You should have an unordered list with three items.
+The unordered list should have the class `item-list`.
```
Comment (Dario-DC) [inline frontend-cert/js-projects/lab-favourite-toggler/user-stories.md:4]:
```suggestion
1. You should have an unordered list with three items.
1. The unordered list should have the class `item-list`.
```
Are these two separate user stories?

File: frontend-cert/js-projects/lab-favourite-toggler/user-stories.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+The three list items should contain the item name followed by a `span` element with the class `favorite-icon ` and an id of `favorite-icon-1`, `favorite-icon-2`, and `favorite-icon-3` respectively.
```
Comment (Dario-DC) [inline frontend-cert/js-projects/lab-favourite-toggler/user-stories.md:6]:
```suggestion
1. The three list items should contain the item name followed by a `span` element with the class `favorite-icon` and an id of `favorite-icon-1`, `favorite-icon-2`, and `favorite-icon-3`, respectively.
```

File: frontend-cert/js-projects/lab-favourite-toggler/user-stories.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+The `span` element should contain the code `&#9825;` initially to represent an empty heart.
```
Comment (Dario-DC) [inline frontend-cert/js-projects/lab-favourite-toggler/user-stories.md:8]:
```suggestion
1. The `span` element should contain the code `&#9825;` initially to represent an empty heart.
```

File: frontend-cert/js-projects/lab-favourite-toggler/user-stories.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+When the `span` element is clicked, the class `filled` should be added to the `span` with class `favorite-icon` if it doesn't exist, and removed if it does.  `&#9825;` should change to `&#10084;` to represent a filled heart.
```
Comment (Dario-DC) [inline frontend-cert/js-projects/lab-favourite-toggler/user-stories.md:10]:
```suggestion
1. When the `span` element is clicked, the class `filled` should be added to the `span` with class `favorite-icon` if it doesn't exist, and removed if it does.  `&#9825;` should change to `&#10084;` to represent a filled heart.
```

File: frontend-cert/js-projects/lab-favourite-toggler/styles.css [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+.favorite-icon {
+    font-size: 20px;
+    cursor: pointer;
+    /* color: gray; */
+}
+
+.favorite-icon.filled {
+    /* color: black; */
```
Comment (Dario-DC) [inline frontend-cert/js-projects/lab-favourite-toggler/styles.css:36]:
```suggestion
```
Can these be removed?

Comment (zairahira) [reply frontend-cert/js-projects/lab-favourite-toggler/styles.css]: done!

**Comment by zairahira** [comment]
It's working fine without `DOMContentLoaded`. 
Would you suggest that I add it?

**Comment by Dario-DC** [comment]
> It's working fine without `DOMContentLoaded`. Would you suggest that I add it?

No, I was just asking because I don't know if it's taught.

**Comment by zairahira** [comment]
Seems like `DOMContentLoaded` isn't mentioned in the sheet. I have asked it to be included in the lecture videos.

**Review by moT01 — APPROVED** [review]
This is pretty cool 👍 LGTM 🎉

**Pull request merged**